File: challenges/3-tipos-de-triangulo/javascript/evertoont/tiposTriangulo.js
const argvA = Number(process.argv.slice(2, 3));
const argvB = Number(process.argv.slice(3, 4));
const argvC = Number(process.argv.slice(4, 5));

const sides = [argvA, argvB, argvC].sort((a, b) => a - b).reverse();

const sideA = sides[0];
const sideB = sides[1];
const sideC = sides[2];

if (sideA < 0 || sideB < 0 || sideC < 0) {
  return console.log("NÃO É PERMITIDO VALOR MENOR QUE ZERO (0)");
}

if (sideA >= sideB + sideC) {
  console.log("NAO FORMA TRIANGULO");
} else if (sideA ** 2 === sideB ** 2 + sideC ** 2) {
  console.log("TRIANGULO RETANGULO");
} else if (sideA ** 2 > sideB ** 2 + sideC ** 2) {
  console.log("TRIANGULO OBTUSANGULO");
} else if (sideA ** 2 < sideB ** 2 + sideC ** 2) {
  console.log("TRIANGULO ACUTANGULO");
}

if (sideA === sideB && sideB === sideC) {
  console.log("TRIANGULO EQUILATERO");
} else if (sideA === sideB || sideB === sideC) {
  console.log("TRIANGULO ISOSCELES");
}

File: challenges/3-tipos-de-triangulo/python/emptybrain/tipos_triangulo.py
A, B, C = input("Entrada: ").split(" ")
values = [float(A), float(B), float(C)]
values.sort(reverse=True)
A = values[0]
B = values[1]
C = values[2]

IVALID: bool = A >= (B + C)
RETANGULO: bool = A**2 == (B**2 + C**2)
OBTUSANGULO: bool = A**2 > (B**2) >= (C**2)
ACUTANGULO: bool = A**2 < (B**2 + C**2)
EQUILATERO: bool = A == B == C
ISOSCELES: bool = A == B != C or A != B == C or A == C != B

if IVALID is True:
    print("NAO FORMA TRIANGULO")
else:
    if OBTUSANGULO is True:
        print("TRIANGULO OBTUSANGULO")

    if ACUTANGULO is True:
        print("TRIANGULO ACUTANGULO")

    if EQUILATERO is True:
        print("TRIANGULO EQUILATERO")

    if ISOSCELES is True:
        print("TRIANGULO ISOSCELES")
